viam-cartographer/src/carto_facade/util.cc

```diff
@@ -56,52 +62,61 @@ int read_pcd(std::string pcd, pcl::PCLPointCloud2 &blob) {
                            data_type, data_idx);
     if (res != 0) {
         LOG(ERROR) << "Failed to parse header";
-    } else {
-        switch (data_type) {
-            // ascii
-            case 0:
-                pcd_stream.seekg(data_idx);
-                res = p.readBodyASCII(pcd_stream, blob, pcd_version);
-                break;
-            // binary
-            case 1: {
-                // This block exists b/c otherwise
-                // map is counterintuitively visible
-                // to the rest of the case
-                // statement branches
-                std::size_t expected_size = data_idx + blob.data.size();
-                std::size_t size = pcd.length();
-                if (expected_size > size) {
-                    LOG(ERROR) << "Corrupted PCD file. The file is smaller "
-                                  "than expected! Expected: "
-                               << expected_size << "actual: " << size;
-                    return -1;
-                }
-                const unsigned char *map =
-                    reinterpret_cast<const unsigned char *>(pcd.c_str());
-                res = p.readBodyBinary(map, blob, pcd_version, false, data_idx);
-            } break;
-            // binary compressed
-            case 2:
-
-                LOG(ERROR) << "compressed PCDs are not currently supported";
-                res = -1;
-                break;
-            default:
-                LOG(ERROR) << "PCD classified as an unsupported data type: "
-                           << data_type;
-                res = -1;
-        }
+        return res;
+    } else if (blob.data.size() == 0) {
```

Comment:
one more nit: this can be just `if` rather than `else if`

Reply:
done